### PR TITLE
Improve grappler layout optimizer to support 5D tensors [Resubmit]

### PR DESCRIPTION
**Pull request by kaixih**
This is for resubmitting #44270.

@nluehr @andyly

### CODE REVIEW

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 1]
```diff
@@ -1848,18 +1960,17 @@ string GetDeviceName(const VirtualPlacer* virtual_placer, const NodeDef& node) {
...
+           "FusedConv2DBiasActivation", "MaxPool", "SpaceToDepth"});
   return default_layout_sensitive_ops->find(node.op()) !=
          default_layout_sensitive_ops->end();
 }
 
 bool IsLayoutSensitiveOp(const NodeDef& node) {
   return IsDefaultLayoutSensitiveOp(node) || IsAvgPoolGrad(node) ||
-         IsBiasAddGrad(node) || IsConv2DBackpropFilter(node) ||
-         IsConv2DBackpropInput(node) ||
+         IsBiasAdd(node) || IsBiasAddGrad(node) ||
```
Comment (andyly) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1972]: This should be restricted to `BiasAdd` also and not include `BiasAddV1`.

We can also expose a function for checking for `BiasAdd` explicitly, and that can be named `IsBiasAddV2`.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer_factory.cc [hunk 2]
```diff
@@ -27,6 +27,9 @@ std::shared_ptr<Transposer> TransposerFactory::GetTransposer(
     return GetOrCreateIfNotFound<DefaultLayoutSensitiveOpTransposer>(
         "DefaultLayoutSensitiveOp");
   }
+  if (IsBiasAdd(node)) {
```
Comment (andyly) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer_factory.cc:30]: This should be restricted to `BiasAdd` also and not include `BiasAddV1`.

And can this be moved to below AvgPoolGrad?

**Review by andyly — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc (hunk 1 above)
Comment (kaixih) [reply tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc]: Done. PTAL.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer_factory.cc (hunk 2 above)
Comment (kaixih) [reply tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer_factory.cc]: Done. PTAL.

**Review by andyly — APPROVED** [review]

**Pull request merged**